ido/src/modules/project.rs

```diff
@@ -352,17 +354,157 @@ impl IDOContract {
+                let mut project_account = self.internal_get_account_by_project_or_panic(project_id,&account_id);
+                let mut project = self.internal_get_project_or_panic(project_id);
+                let account_sale = project_account.sale_data.unwrap();
+                
+                match account_sale.sale_data{ 
+                    AccountSaleData::Lottery(
+                        LotteryAccountSaleData{
+                            eligible_tickets,
+                            deposit_tickets,
+                            ticket_ids,
+                            win_ticket_ids
+                        }) => {
+                            let tickets_num = (deposit / allocation_per_ticket) as u128;
+                            let mut ticket_ids = ticket_ids.clone();
+                            assert!(tickets_num>0, "Must deposit at least {} for exchange a ticket",allocation_per_ticket);
+                            assert!((tickets_num as u64 + deposit_tickets)<=eligible_tickets,"Eligible tickets not enough");
+
+                            if deposit > (tickets_num * allocation_per_ticket){
+                                //  Transfer back change deposit
+                                Promise::new(account_id.clone()).transfer(deposit - tickets_num * allocation_per_ticket);
+                            };
+
+                            let project_total_tickets = (project.total_fund_committed / allocation_per_ticket) as u64;
+                            for i in project_total_tickets..(project_total_tickets+ tickets_num as u64) {
+                                ticket_ids.push(i);
+                            }
+
+                            let lottery_account_sale_data = AccountSaleData::Lottery(
+                                LotteryAccountSaleData{
+                                    eligible_tickets: eligible_tickets,
+                                    deposit_tickets: (deposit_tickets + tickets_num as u64),
+                                    ticket_ids: ticket_ids,
+                                    win_ticket_ids: win_ticket_ids
+                                }  
+                            );
+
+                            let account_sale = AccountSale{
+                                committed_amount: account_sale.committed_amount + (tickets_num*allocation_per_ticket),
+                                sale_data: lottery_account_sale_data
+                            };
+                            
+                            // Update Project account
+                            project_account = ProjectAccount{
+                                sale_data: Some(account_sale),
+                                distribution_data: None
+                            };
+
+                            // Update Project total_fund_committed
+                            project.total_fund_committed += tickets_num*allocation_per_ticket;
```

Comment:
Please also update the total tickets of the project on the variable of `project.lottery_sale_type.total_tickets`.